pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java

```diff
@@ -92,13 +96,17 @@
 public class MergeRollupTaskGenerator implements PinotTaskGenerator {
   private static final int DEFAULT_MAX_NUM_RECORDS_PER_TASK = 50_000_000;
   private static final String REFRESH = "REFRESH";
+  private static final String MERGE_ROLLUP_TASK_DELAY_IN_NUM_BUCKETS = "mergeRollupTaskDelayInNumBuckets";
   private static final Logger LOGGER = LoggerFactory.getLogger(MergeRollupTaskGenerator.class);
 
+  // tableNameWithType -> mergeLevel -> watermarkMs
+  private Map<String, Map<String, Long>> _mergeRollupWatermarks;
   private ClusterInfoAccessor _clusterInfoAccessor;
 
   @Override
   public void init(ClusterInfoAccessor clusterInfoAccessor) {
     _clusterInfoAccessor = clusterInfoAccessor;
+    _mergeRollupWatermarks = new ConcurrentHashMap<>();
```

Comment:
We don't need a concurrent map because all the task generators are single-threaded

Reply:
fixed